### ISSUE TITLE
[BUG]  RX 7900 XTX (gfx1100) stuck in low-power state under compute/ROCm load on linux-cachyos 7.0.11/7.0.12 — works fine on linux-cachyos-lts 6.18.33

### ISSUE THREAD

**Issue by sognix** [CachyOS/linux-cachyos#888]
### Pre-flight checklist

- [x] I have searched existing issues and this is not a duplicate.
- [x] I have read the [Contributing Guidelines](https://github.com/CachyOS/linux-cachyos/blob/master/CONTRIBUTING.md).
- [x] I have verified the issue is reproducible with the latest available CachyOS kernel.
- [x] I have tried to reproduce the issue on Arch Linux's `linux` kernel.

### Upstream / vanilla kernel check

I have not tested with a vanilla/upstream kernel

### Kernel variant

linux-cachyos (EEVDF, Clang)

### System information (cachyos-bugreport.sh)

```text
https://paste.cachyos.org/p/85fb1b1.log
```

### Manual system information (if cachyos-bugreport.sh is unavailable)

```text

```

### Problem Description

After upgrading from an earlier kernel to linux-cachyos 7.0.11-1 (and confirmed still present on 7.0.12-1), the GPU no longer boosts clocks under compute (ROCm/KFD) workloads, while 3D/gaming workloads boost normally.

**Observed behavior**

- Loading a model onto the GPU via PyTorch/ROCm (tested via InvokeAI and a separate HuggingFace transformers fine-tuning script) is dramatically slower than expected — roughly two orders of magnitude. As one data point, InvokeAI's own log reported a single pipeline sub-component (CLIP text encoder, ~234MB) taking 122 seconds to load onto the GPU, where this normally completes in well under a second. The exact total VRAM footprint of the full pipeline was not measured, but the slowdown is consistent and severe across every compute workload tested.
```[2026-06-20 05:28:03,911]::[ModelManagerService]::INFO --> [MODEL CACHE] Loaded model '76120fb2-df02-4247-a12b-9f0124b5a85e:text_encoder' (CLIPTextModel) onto cuda device in 122.34s. Total model size: 234.72MB, VRAM: 234.72MB (100.0%) [2026-06-20 05:28:03,953]::[ModelManagerService]::INFO --> [MODEL CACHE] Loaded model '76120fb2-df02-4247-a12b-9f0124b5a85e:tokenizer' (CLIPTokenizer) onto cuda device in 0.00s. Total model size: 0.00MB, VRAM: 0.00MB (0.0%)```
- Gaming workloads (tested with a native Vulkan title) boost GPU clocks normally and run at expected performance.
- rocm-smi --showclocks during compute load shows:
```
  WARNING: AMD GPU device(s) is/are in a low-power state. Check power control/runtime_status
  GPU[0] : sclk clock level: 1: (227-293Mhz)
  GPU[0] : mclk clock level: 1: (456Mhz)
```
(Expected boost clocks for this card under load: sclk >2000MHz, mclk ~2500MHz)

**Troubleshooting already performed (all ineffective)**

1. power_dpm_force_performance_level set to auto (default) — no boost under compute load.
2. pp_power_profile_mode switched to COMPUTE (index 5) via both LACT GUI and direct sysfs write — confirmed active (* marker correctly shown next to COMPUTE in cat pp_power_profile_mode), but clocks remained low under actual compute workload.
3. power_dpm_force_performance_level set to manual, then pp_dpm_sclk/pp_dpm_mclk written to force the highest available P-state — write is accepted without error (exit 0), but rocm-smi --showclocks continues to report a low clock level (e.g. level 1) instead of the forced level. The SMU appears to silently reject the requested state, similar to the report below.
4. LACT "Power States" — disabling lower P-states, leaving only the two highest enabled — accepted by the driver (confirmed via cat pp_power_profile_mode / power states UI), but actual measured clocks during compute load remained low (and disagreed with what KDE System Monitor displayed at the same time, suggesting the forced range is not actually being honored at the hardware level either).

**Key observation: GFX-ring vs Compute-ring divergence**
Boosting works correctly when load comes through the 3D/GFX ring (gaming), but not when load comes through the compute ring (KFD/ROCm). This points to a regression in the workload-type-to-SMU-hint path for compute submissions specifically, rather than a general DPM/SMU failure (since DPM does respond correctly to GFX-ring workloads).

**Related report**
A very similar symptom (pp_dpm_sclk state masking silently failing on linux-cachyos 7.0.9-1, SMU returning ret=255, write accepted by kernel but rejected by SMU) was reported for a different GPU generation (Polaris/GFX8, RX 580) on the CachyOS forum about a month ago:
https://discuss.cachyos.org/t/amdgpu-pp-dpm-sclk-state-masking-silently-fails-on-polaris-gfx8-smu-returns-ret-255/30115

This suggests the underlying sysfs DPM-state-masking/enforcement path may be broken generically in the current 7.0.x branch, affecting multiple GPU generations, not just RDNA3.

The ArchWiki AMDGPU page also currently notes that manual powerplay table edits are "bugged on kernel side" for Navi3x/Navi4x, which may be related.

### Steps to reproduce

1. Boot linux-cachyos 7.0.11-1 or 7.0.12-1 with an RX 7900 XTX.
2. Run any ROCm/PyTorch compute workload (e.g. load a model with transformers/diffusers, or use InvokeAI for image generation).
3. Monitor rocm-smi --showclocks during the workload.
4. Observe sclk/mclk remaining at the lowest P-state despite active compute load, and the overall workload taking ~100x longer than expected.
5. Reboot into linux-cachyos-lts 6.18.33-2, repeat steps 2–4 — clocks boost correctly and performance is normal.

### Expected behavior

GPU should boost sclk/mclk appropriately under ROCm/compute load, the same way it already does under 3D/gaming load.

### Actual behavior

GPU remains stuck at the lowest available P-state (sclk ~227-293MHz, mclk ~456MHz) for the entire duration of the compute workload, regardless of which power_dpm_force_performance_level mode or pp_power_profile_mode is selected. rocm-smi reports a persistent "low-power state" warning throughout. As a result, GPU-bound compute operations (model loading, inference, training) run roughly 100x slower than expected — a ~234MB model load that should take well under a second instead takes ~122 seconds. The GPU never transitions to a higher P-state during the entire workload, even though the same hardware boosts correctly within seconds under a 3D/gaming workload.

### Logs / stack traces

```text

```

### Additional system information

- dmesg shows normal amdgpu/KFD initialization with no errors (KFD device created successfully, kfd kfd: added device 1002:744c).
- The standard SMU driver if version not matched message is present in dmesg but appears to be benign/expected on this hardware generation (seen in other working RX 7900 XTX dmesg logs as well).
- Happy to provide full dmesg output, rocm-smi -a output, or test additional kernel parameters/patches if it helps narrow this down.
- I have checkted the Arch Linux Kernel box, but only because it is mandatory. I have NOT tested it, because
setting up a separate vanilla Arch install solely to test this wasn't practical (and I don't have an additional system for it); happy to help narrow this down another way if useful.

**Comment by waszak** [comment]
There is also some huge performance regression in ComfyUI LoadCheckpoint on 7.0.12
Idk if its connected to powerstate but like i am linking this maybe its connected somehow

https://github.com/Comfy-Org/ComfyUI/issues/14488

.EDIT Try Kernel 7.1.1 from what i tested it works much better than 7.0.12.
Solved some issues

**Comment by superm1** [comment]
I suggest bisecting this issue to confirm exactly what commit caused it.

**Comment by paeddym** [comment]
Something definitely seems to have broken regarding the power reporting for the 7900XTX. In LACT the exposed power limit on CachyOS 7.1.1-2 is 334W, while before updating last night it used to be over 400W for my model of the card

<img width="1108" height="757" alt="Image" src="https://github.com/user-attachments/assets/d0a3c41e-8d62-4086-8314-7e63bb90ab88" />

**Comment by sognix** [comment]
> There is also some huge performance regression in ComfyUI LoadCheckpoint on 7.0.12 Idk if its connected to powerstate but like i am linking this maybe its connected somehow
> 
> [Comfy-Org/ComfyUI#14488](https://github.com/Comfy-Org/ComfyUI/issues/14488)
> 
> .EDIT Try Kernel 7.1.1 from what i tested it works much better than 7.0.12. Solved some issues


I can confirm the kernel. After today's update, it's working again as far as I can tell. I only did a quick test, though.

**Comment by Flock82** [comment]
Same issue with Power Limit reduced from ~400W to 334 W (which is below the default setting on 350 W) on a Sapphire 7900 XTX and Cachy Kernel 7.1.1-2
Also needed to manually adjust the LACT config.yml, as the previous value is reported "out of range", crashing the LACT GUI.

**Comment by FerrumMaster** [comment]
It might the device load order. There still is this bug that related to this.

https://gitlab.freedesktop.org/drm/amd/-/work_items/2356

And if I do list my GPU it is at the end now, never was and power setting adjustment returns error as initial value is reported zero.

for i in /sys/class/hwmon/hwmon*; do echo "$i: $(cat $i/name)"; done
/sys/class/hwmon/hwmon0: nvme
/sys/class/hwmon/hwmon1: nvme
/sys/class/hwmon/hwmon2: nvme
/sys/class/hwmon/hwmon3: nvme
/sys/class/hwmon/hwmon4: k10temp
/sys/class/hwmon/hwmon5: r8169_0_a00:00
/sys/class/hwmon/hwmon6: r8169_0_b00:00
/sys/class/hwmon/hwmon7: i915
/sys/class/hwmon/hwmon8: nct6798
/sys/class/hwmon/hwmon9: amdgpu

grep -s . /sys/class/drm/card0/device/hwmon/hwmon9/power1_cap
0

For the sake of experiment I nuked the OS. Stock Fedora kernels also have this bug. Downgrading 6.19 kernel I had fixed the issue. 7.1.3 and 7.1.2 have it.

It seems this is our antihero dev.

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v7.0.13&id=f02262d654f9ffdc83a75eb35f1d94766c52f6e2

**Comment by SoongVilda** [comment]
This is confirmed on stock Fedora kernels and is already tied to the upstream DRM/AMD work item. Closing the CachyOS copy to keep one source of truth; please reopen only for a specific upstream fix that needs backporting.
